### PR TITLE
Content service sequence diagram

### PR DESCRIPTION
**Pull request by tisnik**
# Description

Content service sequence diagram

Fixes https://projects.engineering.redhat.com/browse/CCXDEV-2313

## Type of change

- Documentation update

## Testing steps

N/A

### CODE REVIEW

**Comment by joselsegura** [comment]
It could be a bit weird, but probably it is worthy to include "Insights Content Service image generation", because with the current diagram it looks like the service, after being restarted, is cloning the repository, and we only do that when generating the container image, not on every restart.

The Smart Proxy is retrieving periodically all the content from Content Service. With the current diagram it looks like content is requested on demand

**Comment by tisnik** [comment]
> It could be a bit weird, but probably it is worthy to include "Insights Content Service image generation", because with the current diagram it looks like the service, after being restarted, is cloning the repository, and we only do that when generating the container image, not on every restart.
> 
> The Smart Proxy is retrieving periodically all the content from Content Service. With the current diagram it looks like content is requested on demand

Good observations, thank you @joselsegura

File: docs/architecture/content_service_seq_diagram.uml [hunk 1]
```diff
@@ -29,26 +29,34 @@ footer Copyright © 2020 Red Hat, Inc. Author: Pavel Tisnovsky
...
+openshift -> github:  Clone\nrepository
+github -> openshift: Repository content
+openshift -> openshift: Rebuild image
+note right: OpenShift rebuild Insights\nContent Service image\nwith new rules content included
+openshift -> openshift: Restart pod
+openshift -> content_service: Set replicas > 0
+note right: Now the Content Service\nexposes new rules content\nvia its REST API
+
+== Synchronization between Smart Proxy and Content Service ==
+smart_proxy -> content_service: Get content\nfor rule ID
```
Comment (joselsegura) [inline docs/architecture/content_service_seq_diagram.uml:53]: I would change the text of this line: smart-proxy update **all** the content at a time, not only for a given rule ID as can be understood in the current diagram.

Comment (tisnik) [reply docs/architecture/content_service_seq_diagram.uml:53]: sure, will fix it!

**Pull request merged**